lib/ruby_smb/dcerpc/icpr.rb

```diff
@@ -72,7 +72,7 @@ def cert_server_request(attributes:, authority:, csr:)
               status_code: status_code
             )
           end
-        else
+        elsif !cert_server_request_response.pctb_encoded_cert.buffer.empty?
```

Comment:
Doesn't look like we are checking if `cert_server_request_response` is `nil` here? Not sure if that check should be moved to be earlier in the code though. Same also applies to `pctb_encoded_cert` as well.

Other than that this looks good!

Reply:
It won't be nil, it's defined on L48 and if the read fails it's raised as an InvalidPacket error already. If `cert_server_request_response` is defined, then the `pctb_encoded_cert` attribute is defined because that's in the BinData definition.